src/rmp/src/Restructure.cpp

```diff
@@ -0,0 +1,555 @@
+    for (auto pinName : *error) {
+      debugPrint(logger_, RMP, "remap", 1, "Unclocked pin: {}", pinName);
+      if (!first && open_sta_->getDbNetwork()->findPin(pinName)) {
+        ends.insert(open_sta_->getDbNetwork()->findPin(pinName));
+      }
+    }
+  }
+  logger_->report("Number of end points for restructure {}", ends.size());
+}
+
+int Restructure::countConsts(odb::dbBlock* top_block)
+{
+  int const_nets = 0;
+  for (auto block_net : top_block->getNets()) {
+    if (block_net->getSigType() == odb::dbSigType::GROUND
+        || block_net->getSigType() == odb::dbSigType::POWER)
+      const_nets++;
+  }
+
+  return const_nets;
+}
+
+void Restructure::removeConstCells()
+{
+
+  if (!hicell_.size()  || !locell_.size())
+    return;
+
+  odb::dbMaster* hicell_master = nullptr;
+  odb::dbMaster* locell_master = nullptr;
+
+  for (auto&& lib : block_->getDb()->getLibs()) {
+    hicell_master = lib->findMaster(hicell_.c_str());
+
+    locell_master = lib->findMaster(locell_.c_str());
+    if (locell_master && hicell_master)
+      break;
+  }
+  if (!hicell_master || !locell_master)
+    return;
+
+  open_sta_->ensureGraph();
+  open_sta_->ensureLevelized();
+  open_sta_->searchPreamble();
+
+  std::set<odb::dbInst*> constInsts;
+  int const_cnt = 1;
+  for (auto inst : block_->getInsts()) {
+    int outputs = 0;
+    int const_outputs = 0;
+    for (auto&& iterm : inst->getITerms()) {
+      auto mterm = iterm->getMTerm();
+      auto net = iterm->getNet();
+
+      if (iterm->getSigType() == odb::dbSigType::POWER
+          || iterm->getSigType() == odb::dbSigType::GROUND)
+        continue;
+
+      if (iterm->getIoType() != odb::dbIoType::OUTPUT)
+        continue;
+      outputs++;
+      sta::Vertex *vertex, *bidirect_drvr_vertex;
+      auto pin_ = open_sta_->getDbNetwork()->dbToSta(iterm);
+      open_sta_->getDbNetwork()->graph()->pinVertices(pin_, vertex, bidirect_drvr_vertex);
+      sta::LogicValue pinVal = ((vertex)
+                 ? vertex->simValue()
+                 : ((bidirect_drvr_vertex) ? bidirect_drvr_vertex->simValue()
+                                           : sta::LogicValue::unknown));
+      if (pinVal == sta::LogicValue::one || pinVal == sta::LogicValue::zero) {
+        odb::dbNet* net = iterm->getNet();
+        if (net) {
+          odb::dbMaster* const_master = (pinVal == sta::LogicValue::one) ? hicell_master : locell_master;
+          std::string const_port = (pinVal == sta::LogicValue::one) ? hiport_ : loport_;
+          std::string inst_name = "rmp_const_" + std::to_string(const_cnt);
+          debugPrint(logger_, RMP, "remap", 2, "Adding cell {} inst {} for {}",
+                                      const_master->getName(), inst_name, inst->getName());
+          auto new_inst = odb::dbInst::create(block_, const_master, inst_name.c_str());
+          if (new_inst) {
+            odb::dbITerm::disconnect(iterm);
+            odb::dbITerm::connect(new_inst->findITerm(const_port.c_str()), net);
```

Comment:
It would be cleaner to find the dbMTerm once and use dbInst::getITerm for the mterm rather than using names.